RadioGroup: Add InputClass and InputStyle for backward compatibility

mckaragoz opened the pull request:
<!--
MAKE SURE TO READ THE CONTRIBUTING GUIDE BEFORE CREATING A PR
https://github.com/MudBlazor/MudBlazor/blob/dev/CONTRIBUTING.md

Testing sections can be removed for documentation changes
-->

<!-- Provide a general summary of your changes in the Title above -->
<!-- Keep the title short and descriptive, as it will be used as a commit message -->


## Description
Fixes #4916.

With #4084, class and style parameters moved to another element in the component. Now we added extra InputClass and and InputStyle parameters the previous place.

So if a user already had their custom classes or styles (if its about flex or etc), only need change:

Class => InputClass
Style => InputStyle

The upper radiogroup has Class and below radiogroup has InputClass `d-flex justify-space-between`

![Ekran görüntüsü 2022-07-23 230048](https://user-images.githubusercontent.com/78308169/180621279-7b73713a-1637-4096-971e-bbaa7364ccf3.png)


## How Has This Been Tested?
<!-- All PR's should implement unit tests if possible -->
<!-- Please describe how you tested your changes. -->
<!-- Have you created new tests or updated existing ones? -->
<!-- e.g. unit | visually | none -->

## Types of changes
<!-- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

<!-- If you made any visual changes, provide screenshots of before/after, it its moving parts, please provide high quality gif, wemb or mp4 -->

## Checklist:
<!-- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!-- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [x] The PR is submitted to the correct branch (`dev`).
- [x] My code follows the code style of this project.
- [x] I've added relevant tests.

mckaragoz commented:
@just-the-benno could you review this? KeyInterceptor crash and this boolean input validation PR's had unexpected changes, so when we fix these both, users can upgrade their package safely.

just-the-benno commented on src/MudBlazor/Components/Radio/MudRadioGroup.razor:
```diff
@@ -5,7 +5,7 @@
     <InputContent>
         <CascadingValue Value="@((IMudRadioGroup)this)" IsFixed="true">
-            <div role="radiogroup" @attributes="UserAttributes" >
+            <div role="radiogroup" @attributes="UserAttributes" class="@InputClass" style="@InputStyle">
```
To prevent this kind of happening again, you should also add a new class to it so that the styling can also be done with that class.  Maybe something like ``mud-radio mud-radio-group``. In that case, you would need to use the CssBuilder. 
 ``GetInputClass()``. I've added a description in the next comment.

just-the-benno commented on src/MudBlazor/Components/Radio/MudRadioGroup.razor.cs:
```diff
@@ -21,6 +21,20 @@ public partial class MudRadioGroup<T> : MudFormComponent<T, T>, IMudRadioGroup
+        [Parameter]
+        [Category(CategoryTypes.Radio.Appearance)]
+        public string InputClass { get; set; }
+
```
and in this file you can add the method

```csharp
private String GetInputClass() =>
new CssBuilder("mud-radio mud-radio-group").AddClass(InputClass );
```

just-the-benno reviewed (changes requested):
Your proposal is a good way forward.

Please take the time to add unit tests to ensure that the DOM structure of this component won't be changed again (at least without being aware of it) and that this class and styles are applied correctly.

mckaragoz commented:
@just-the-benno if i understand true, changed input class with a method and added test to check which class is in input control and which class is in inner div.

just-the-benno commented on src/MudBlazor.UnitTests/Components/RadioTests.cs:
```diff
@@ -13,6 +13,25 @@ namespace MudBlazor.UnitTests.Components
+            var inputControl = comp.FindComponent<MudInputControl>();
+            inputControl.Instance.InputContent.Should().NotBeNull();
+
+            comp.Markup.Should().Contain("mud-radio-group");
```
You can remove this line. It is covered by your test on the next line

just-the-benno commented on src/MudBlazor.UnitTests/Components/RadioTests.cs:
```diff
@@ -13,6 +13,25 @@ namespace MudBlazor.UnitTests.Components
+            comp.FindAll(".some-main-class .some-input-class").Count.Should().Be(1);
+            comp.FindAll(".mud-radio").Count.Should().Be(3);
+            // Input content should not have main class (Classname), but should have input class (InputClass)
+            comp.Markup.Should().NotContain("mud-radio-group some-main-class");
```
try to write it from the component part of view.  And

```csharp
comp.FindAll(".mud-radio-group.some-main-class").Should().BeEmpty();
```

just-the-benno commented on src/MudBlazor.UnitTests/Components/RadioTests.cs:
```diff
@@ -13,6 +13,25 @@ namespace MudBlazor.UnitTests.Components
+            comp.Markup.Should().Contain("mud-radio-group");
+            comp.FindAll("div.mud-radio-group").Count.Should().Be(1);
+            comp.FindAll("div.some-main-class").Count.Should().Be(1);
+            comp.FindAll("div.some-input-class").Count.Should().Be(1);
```
Just a note: You could also use fluent assertion's "collection" capabilities. 

```csharp
comp.FindAll("div.some-input-class").Should().ContainSingle()
```

just-the-benno reviewed (approved):
Hi, @mckaragoz thanks for the changes and the added test.

If you have time, I made some notes on how you could improve the tests, but that is optional.  Nothing needs to be changed. It is ready to be merged

mckaragoz commented:
@henon could you look at this?

henon commented:
Thanks!
